### PR TITLE
NW6 | Rabia Avci | Full-Stack-Project | Week-2 | Post new video endpoint

### PR DESCRIPTION
**Pull request by RbAvci**
Completed these tasks for this ticket:

-  There is a POST /api/videos backend endpoint to facilitate adding videos to the database.
-  The endpoint generates and attaches a unique ID to each newly added video.
-  The endpoint accepts video data in JSON format (example provided).
-  The endpoint responds with a JSON response that includes the generated id (example provided)

### CODE REVIEW

**Review by zelihapala — APPROVED** [review]
Well done!

File: db/initdb.sql [hunk 1]
```diff
@@ -3,8 +3,10 @@
 DROP TABLE IF EXISTS videos CASCADE;
 
 CREATE TABLE videos (
+    id SERIAL PRIMARY KEY,
```
Comment (Ara225) [inline db/initdb.sql:6]: 👍

File: server/api.js [hunk 2]
```diff
@@ -13,8 +13,18 @@ router.get("/videos", async (_, res) => {
...
 });
 
-router.get("/health", async (_, res) => {
-	res.sendStatus(200);
+router.post("/videos", async (req, res) => {
+	if (!req.body.title) {
+		return res.status(422).json({ message: "Title field is required" });
+	}
+	if (!req.body.src) {
+		return res.status(422).json({ message: "src field is required" });
```
Comment (Ara225) [inline server/api.js:21]: It is good to see you using HTTP status codes other than 200 and thinking carefully about the right one to use. However, it's worth being aware that there are conventions - 400 bad request is usually  used for this situation. It makes it very clear that the problem is on the users side.

The message is a nice touch and would be much appreciated when debugging!

File: server/api.js [hunk 3]
```diff
@@ -13,8 +13,18 @@ router.get("/videos", async (_, res) => {
...
+	}
+	const result = await db.query(
+		`INSERT INTO videos (title,src) VALUES ('${req.body.title}','${req.body.src}') RETURNING id`
```
Comment (Ara225) [inline server/api.js:24]: This line works, but sadly it's  horribly insecure thanks to SQL injection. Anyone could delete your videos table just by sending a src with a value like this:
Fakesrc');DROP TABLE videos;

I don't know if Kieth mentioned this, and even if he did, it's easy not to make the connection so there’s no shame in having made the mistake. However, you need to get into the habit of doing  the right thing have a look at  https://stackoverflow.com/questions/58174695/prevent-sql-injection-with-nodejs-and-postgres

File: server/api.js [hunk 4]
```diff
@@ -13,8 +13,18 @@ router.get("/videos", async (_, res) => {
...
+	);
+	const newVideoId = result.rows[0].id;
+	res.status(200).json({ success: true, data: { id: newVideoId } });
```
Comment (Ara225) [inline server/api.js:27]: It's good that you're sending  back the video ID that's  extremely useful 👍

File: server/app.js [hunk 5]
```diff
@@ -6,11 +6,29 @@ import { fileURLToPath } from "node:url";
...
+		return res.status(400).json({ message: "Invalid JSON format" });
+	}
+	next(err);
+});
+
+// General error handler
+app.use((err, req, res, next) => {
+	res.status(500).json({ message: "An unexpected error occurred" });
+	next(err);
+});
```
Comment (Ara225) [inline server/app.js:30]: So good to see error handling and the health endpoint but there's a major issue with doing it just in this file. None of the stuff in this file ever runs in Netifly (the starting point for Netlifly is functions/app.mjs), so this code only works on your local machine. If this is intentional, that's  fine, but please add a comment to clarify :)

**Review by Ara225 — CHANGES_REQUESTED** [review]
Overall, this works just fine, but there are some important issues that are worth reviewing. You'll find that as your applications get more complex, there are more of these issues. Embrace them all learn the why, they're intresting and will help you grow your  knowledge

**Pull request merged**